client/data/courses/use-update-user-course-progression-mutation.js

```diff
@@ -18,9 +18,9 @@ function useUpdateUserCourseProgressionMutation( queryOptions = {} ) {
 			),
 		{
 			...queryOptions,
-			onSuccess( ...args ) {
-				queryClient.invalidateQueries( [ 'course-progression', args.courseSlug ] );
-				queryOptions.onSuccess?.( ...args );
+			onSuccess( data, variables, context ) {
+				queryClient.invalidateQueries( [ 'courses', variables.courseSlug ] );
```

Comment:
Invalidate the query to fetch the latest data so that we don't need to handle local completed status.

Reply:
This change looks good. If feels like the `POST` could return the new progression data so we can update the local cache without using another `GET`. This works though.
I'm a bit confused about how it was working before. I can't see what the old `[ 'course-progression', args.courseSlug ]` query key was referring to.